### PR TITLE
is_subclass_of do not infer anything when context is falsy

### PR DESCRIPTION
**Pull request by VincentLanglet**
Closes https://github.com/phpstan/phpstan/issues/6891

### CODE REVIEW

**Comment by ondrejmirtes** [comment]
Regression test for #6891 needs to be in NodeScopeResolverTest.

**Comment by VincentLanglet** [comment]
> Regression test for #6891 needs to be in NodeScopeResolverTest.

Done

**Comment by herndlm** [comment]
👍
does it make sense to do the same for `is_a`? could be done in a follow-up as well of course
UPDATE: checked this now, apparently this is not needed for is_a and would even break things hmm

File: src/Type/Php/IsSubclassOfFunctionTypeSpecifyingExtension.php [hunk 1]
```diff
@@ -34,18 +33,15 @@ public function isFunctionSupported(FunctionReflection $functionReflection, Func
 
 	public function specifyTypes(FunctionReflection $functionReflection, FuncCall $node, Scope $scope, TypeSpecifierContext $context): SpecifiedTypes
 	{
-		if (count($node->getArgs()) < 2) {
+		if (!$context->truthy() || count($node->getArgs()) < 2) {
```
Comment (staabm) [inline src/Type/Php/IsSubclassOfFunctionTypeSpecifyingExtension.php:36]: This `$context->truthy()` check should be done in `isFunctionSupported` instead

**Comment by ondrejmirtes** [comment]
Thank you.

**Pull request merged**